### PR TITLE
Feature rewards gambling 2

### PR DESCRIPTION
**Pull request by AlexandreMajchrzak**
look at #392 please. I got false banned midway...

### CODE REVIEW

File: app/src/main/java/com/github/se/cyrcle/model/user/User.kt [hunk 1]
```diff
@@ -66,3 +66,31 @@ data class User(
...
+          80 to LevelRange("ლ", "#8B0000"), // Dark Red
+          90 to LevelRange("☤", "#B8860B"), // Dark Gold (Goldenrod)
+          100 to LevelRange("♔", "rainbow") // Special case for rainbow effect
+          )
+
+  /** Gets the appropriate level range properties for a given score */
+  fun getLevelRange(score: Double): LevelRange {
+    val level = score.toInt()
+    val rangeStart = ((level / 10) * 10).coerceAtMost(100)
+    return levelRanges[rangeStart] ?: LevelRange("⭒", "#2B2B2B") // Default fallback
```
Comment (Armetiis) [inline app/src/main/java/com/github/se/cyrcle/model/user/User.kt:94]:
```suggestion
    return levelRanges[rangeStart] ?: levelRanges[0] // Default fallback
```

File: app/src/main/java/com/github/se/cyrcle/ui/gambling/GamblingScreen.kt [hunk 2]
```diff
@@ -281,20 +330,98 @@ fun GamblingScreen(navigationActions: NavigationActions, userViewModel: UserView
...
+                                      .testTag("xp_increment_text"))
+                        }
+                  }
 
               WheelView(
                       modifier = Modifier.size(300.dp).testTag("wheel_view"),
-                      onSegmentLanded = { /* Handle landing */})
+                      onSegmentLanded = { segmentName ->
+                        val reputationIncrement =
+                            when (segmentName) {
```
Comment (Armetiis) [inline app/src/main/java/com/github/se/cyrcle/ui/gambling/GamblingScreen.kt:392]: Here to avoid "code duplication" you could create a val for the normalization factor `sqrt(currentLevel + 1.0)`

File: app/src/androidTest/java/com/github/se/cyrcle/ui/gambling/GamblingScreenTest.kt [hunk 3]
```diff
@@ -158,7 +199,25 @@ class GamblingScreenTest {
...
-    composeTestRule.mainClock.advanceTimeBy(10000)
+    // Wait for second spin to complete
+    run {
+      val timeoutMs = 15000L
+      val incrementMs = 1000L
+      var elapsedTime = 0L
+
+      while (elapsedTime < timeoutMs) {
+        composeTestRule.mainClock.advanceTimeBy(incrementMs)
+        Thread.sleep(1000)
```
Comment (LouisJamesVasseur) [inline app/src/androidTest/java/com/github/se/cyrcle/ui/gambling/GamblingScreenTest.kt:210]: Does the Thread.sleep need to be here?

File: app/src/androidTest/java/com/github/se/cyrcle/ui/gambling/GamblingScreenTest.kt
Comment (LouisJamesVasseur) [inline app/src/androidTest/java/com/github/se/cyrcle/ui/gambling/GamblingScreenTest.kt:1]: In general, there's a lot of constants in this file. Would be nice if you could refactor them into const vals, and remove the Thread.sleeps if at all possible

File: app/src/main/java/com/github/se/cyrcle/ui/gambling/GamblingScreen.kt [hunk 4]
```diff
@@ -181,10 +190,8 @@ fun WheelView(
           rotation = targetRotation % 360
           isSpinning = false
           pauseAfterSpin = true
-          delay(2000)
```
Comment (LouisJamesVasseur) [inline app/src/main/java/com/github/se/cyrcle/ui/gambling/GamblingScreen.kt:184]: Good work removing the delays!

File: app/src/main/java/com/github/se/cyrcle/ui/gambling/GamblingScreen.kt [hunk 5]
```diff
@@ -209,11 +216,9 @@ fun WheelView(
 
           val startAngle = index * segmentAngle
           val middleAngle = Math.toRadians((startAngle + segmentAngle / 2.0))
-          val textRadius = radius * 0.65f // Moved closer to edge (was 0.6f)
+          val textRadius = radius * 0.65f
```
Comment (LouisJamesVasseur) [inline app/src/main/java/com/github/se/cyrcle/ui/gambling/GamblingScreen.kt:219]: Maybe a const val here, again?

File: app/src/main/java/com/github/se/cyrcle/ui/parkingDetails/ParkingDetailsScreen.kt [hunk 6]
```diff
@@ -460,15 +478,30 @@ fun ParkingDetailsScreen(
...
+                                    text =
+                                        stringResource(
+                                            R.string.display_user_tag_format,
+                                            range.symbol,
+                                            level,
+                                            ownerUsername),
+                                    style = MaterialTheme.typography.bodyMedium,
+                                    color =
+                                        MaterialTheme.colorScheme
+                                            .onSurface // TODO rainbow text color
```
Comment (Armetiis) [inline app/src/main/java/com/github/se/cyrcle/ui/parkingDetails/ParkingDetailsScreen.kt:492]: I see that there are way too many places in the code with an if else on the color argument. Here is my suggestion:
You create a composable in atoms called DisplayText that takes as argument a composable Text and a string representing the color tag. DisplayText will be the one in charge of processing the string input and discriminate betweeen "rainbow" or a hex color tag and this will avoid having and If ("rainbow") ... else ... in all three files that display the username.. I don't know if that was clear. If it wasn't, please let me know. It's ok if for now the color isn't displayed as rainbow. The real problem is that this if else makes the code hard to read and maintain

File: app/src/androidTest/java/com/github/se/cyrcle/ui/gambling/GamblingScreenTest.kt (hunk 3 above)
Comment (AlexandreMajchrzak) [reply app/src/androidTest/java/com/github/se/cyrcle/ui/gambling/GamblingScreenTest.kt:210]: yes it does

Comment (LouisJamesVasseur) [reply app/src/androidTest/java/com/github/se/cyrcle/ui/gambling/GamblingScreenTest.kt:210]: Why is that?

File: app/src/androidTest/java/com/github/se/cyrcle/ui/profile/ViewProfileScreenTest.kt [hunk 7]
```diff
@@ -220,7 +220,7 @@ class ViewProfileScreenTest {
     composeTestRule.onNodeWithTag("EditButton").assertIsDisplayed()
     composeTestRule.onNodeWithTag("DisplayFirstName").assertTextEquals("Jane")
     composeTestRule.onNodeWithTag("DisplayLastName").assertTextEquals("Smith")
-    composeTestRule.onNodeWithTag("DisplayUsername").assertTextEquals("@janesmith")
+    composeTestRule.onNodeWithTag("DisplayUsernameWithLevel").assertTextEquals("[☤92] janesmith")
```
[the remark this answers is not in the document]
Comment (AlexandreMajchrzak) [reply app/src/androidTest/java/com/github/se/cyrcle/ui/profile/ViewProfileScreenTest.kt:223]: These symbols are part of Unicode so there's a good chance that it can display well everywhere. And we're working on an android app anyways for this project so who cares for now.

File: app/src/main/java/com/github/se/cyrcle/ui/parkingDetails/ParkingDetailsScreen.kt [hunk 8]
```diff
@@ -80,19 +84,30 @@ fun ParkingDetailsScreen(
...
   val showDialogImage = remember { mutableStateOf<String?>(null) }
   val showDialogImageDestinationPath = remember { mutableStateOf<String?>("") }
+
+  var ownerReputationScore = 0.0
+  var ownerUsername = stringResource(R.string.undefined_username)
   if (selectedParking.owner != "Unknown Owner" && selectedParking.owner != null) {
-    userViewModel.selectSelectedParkingUser(parkingViewModel.selectedParking.value?.owner!!)
-  } else {
-    userViewModel.setParkingUser(null)
+    userViewModel.getUserById(
```
Comment (LouisJamesVasseur) [inline app/src/main/java/com/github/se/cyrcle/ui/parkingDetails/ParkingDetailsScreen.kt:101]: Why did you remove this line? Pretty sure that field needs to be added still

File: app/src/androidTest/java/com/github/se/cyrcle/ui/gambling/GamblingScreenTest.kt [hunk 9]
```diff
@@ -87,16 +89,55 @@ class GamblingScreenTest {
...
     composeTestRule.onNodeWithTag("coin_display").assertTextEquals("Coins: 25")
+
     // First spin (25 coins -> 15 coins)
     composeTestRule.onNodeWithTag("spin_button").assertIsEnabled()
     composeTestRule.onNodeWithTag("spin_button").performClick()
-    composeTestRule.mainClock.advanceTimeBy(10000)
+
+    // Wait for first spin to complete
+    run {
+      val timeoutMs = 15000L
```
[the remark this answers is not in the document]
Comment (AlexandreMajchrzak) [reply app/src/androidTest/java/com/github/se/cyrcle/ui/gambling/GamblingScreenTest.kt:99]: the wheel being in constant movement, the testing is absolutely horrendous and i have to pause the clock and advance time manually, and after trial and error i came up with all these values you can see in here. A spin wont take more than 15 seconds.

**Review by Armetiis — CHANGES_REQUESTED** [review]
Hi. I think you did a pretty good job overrall. However, for some reason i still can't see the level of other users (e.g. in the image below, gokunaruto... should be level 15 but I see level 0).

Also please mind my comment on the duplication of if (color == "rainbow") ... else ...
These are pretty big pieces of code that are duplicated (three times). I'd be glad to explain further the implementation i have in mind if my comment was not clear enough. 

image

File: app/src/main/java/com/github/se/cyrcle/ui/gambling/GamblingScreen.kt
[the remark this answers is not in the document]
Comment (Armetiis) [reply app/src/main/java/com/github/se/cyrcle/ui/gambling/GamblingScreen.kt:1]: I agree with Louis here!!

File: app/src/androidTest/java/com/github/se/cyrcle/ui/profile/ViewProfileScreenTest.kt (hunk 7 above)
[the remark this answers is not in the document]
Comment (Armetiis) [reply app/src/androidTest/java/com/github/se/cyrcle/ui/profile/ViewProfileScreenTest.kt:223]: I think the choice of Unicode characters was very thoughtful and very minecraft factions of you

[the remark this answers is not in the document]
Comment (LouisJamesVasseur) [reply app/src/androidTest/java/com/github/se/cyrcle/ui/profile/ViewProfileScreenTest.kt:223]: Agreed. Good job making sure that this was taken into account, shows you put some good thought into it!!

File: app/src/androidTest/java/com/github/se/cyrcle/ui/gambling/GamblingScreenTest.kt (hunk 3 above)
Comment (AlexandreMajchrzak) [reply app/src/androidTest/java/com/github/se/cyrcle/ui/gambling/GamblingScreenTest.kt:210]: Good question, i dont have the answer but the test doesnt pass without, or with a lower value...

File: app/src/androidTest/java/com/github/se/cyrcle/ui/profile/ViewProfileScreenTest.kt (hunk 7 above)
[the remark this answers is not in the document]
Comment (AlexandreMajchrzak) [reply app/src/androidTest/java/com/github/se/cyrcle/ui/profile/ViewProfileScreenTest.kt:223]: > I think the choice of Unicode characters was very thoughtful and very minecraft factions of you. SPEAKING OF FACTIONS (just kidding)

I literally took minecraft inspiration. I was on a bunch of lifetime leaderboards on hypixel for years ;) haha

File: app/src/main/java/com/github/se/cyrcle/ui/parkingDetails/ParkingDetailsScreen.kt (hunk 6 above)
Comment (LouisJamesVasseur) [reply app/src/main/java/com/github/se/cyrcle/ui/parkingDetails/ParkingDetailsScreen.kt:492]: Yeah, I was also thinking something along those lines but you beat me to it. It's a Composable that's often repeated in code so it makes sense to use an Atom for this, w/ the only "varying" factor being the color

File: app/src/main/java/com/github/se/cyrcle/model/user/User.kt (hunk 1 above)
Comment (AlexandreMajchrzak) [reply app/src/main/java/com/github/se/cyrcle/model/user/User.kt:94]: you're right, that's cleaner

File: app/src/main/java/com/github/se/cyrcle/ui/profile/ViewProfileScreen.kt [hunk 10]
```diff
@@ -449,7 +449,8 @@ private fun UserReviewsSection(
                       },
                       options = mapOf(),
                       userViewModel = userViewModel,
-                      reviewViewModel = reviewViewModel)
+                      reviewViewModel = reviewViewModel,
+                      ownerReputationScore = null)
```
Comment (LouisJamesVasseur) [inline app/src/main/java/com/github/se/cyrcle/ui/profile/ViewProfileScreen.kt:453]: Why put the ownerReputationScore as null and not simply 0 here?

**Review by LouisJamesVasseur — COMMENTED** [review]
Already mentioned how impressed I was in general w/ the feature, and I'm happy to see that some of my feedback was taken into account already. Please mind the comments I added below

File: app/src/main/java/com/github/se/cyrcle/model/user/User.kt (hunk 1 above)
Comment (AlexandreMajchrzak) [reply app/src/main/java/com/github/se/cyrcle/model/user/User.kt:94]: except that it doesnt work ??

File: app/src/main/java/com/github/se/cyrcle/ui/gambling/GamblingScreen.kt (hunk 2 above)
Comment (AlexandreMajchrzak) [reply app/src/main/java/com/github/se/cyrcle/ui/gambling/GamblingScreen.kt:392]: indeed

File: app/src/main/java/com/github/se/cyrcle/model/user/User.kt (hunk 1 above)
Comment (AlexandreMajchrzak) [reply app/src/main/java/com/github/se/cyrcle/model/user/User.kt:94]: because if i take your suggestion the return type needs to be changed to LevelRange? and so now everywhere i call the function i need to add safe "?" call

File: app/src/main/java/com/github/se/cyrcle/ui/gambling/GamblingScreen.kt (hunk 5 above)
Comment (AlexandreMajchrzak) [reply app/src/main/java/com/github/se/cyrcle/ui/gambling/GamblingScreen.kt:219]: yes why not

File: app/src/main/java/com/github/se/cyrcle/ui/gambling/GamblingScreen.kt [hunk 11]
```diff
@@ -281,20 +330,98 @@ fun GamblingScreen(navigationActions: NavigationActions, userViewModel: UserView
...
+                          Text(
+                              text = stringResource(R.string.level_display, (displayedLevel + 1)),
+                              style = MaterialTheme.typography.bodyMedium,
+                              modifier = Modifier.testTag("next_level_text"))
+                        }
+
+                    LinearProgressIndicator(
+                        progress = { animatedProgress.value },
+                        modifier =
+                            Modifier.fillMaxWidth(0.8f).height(8.dp).testTag("level_progress_bar"),
```
[the remark this answers is not in the document]
Comment (AlexandreMajchrzak) [reply app/src/main/java/com/github/se/cyrcle/ui/gambling/GamblingScreen.kt:352]: by camel case you mean LevelProgressBar instead of snake level_progress_bar ?

[the remark this answers is not in the document]
Comment (LouisJamesVasseur) [reply app/src/main/java/com/github/se/cyrcle/ui/gambling/GamblingScreen.kt:352]: Exactly right!

File: app/src/main/java/com/github/se/cyrcle/ui/profile/ViewProfileScreen.kt (hunk 10 above)
Comment (AlexandreMajchrzak) [reply app/src/main/java/com/github/se/cyrcle/ui/profile/ViewProfileScreen.kt:453]: this was a suggestion from Maria, because in the ViewProfileScreen we dont see the owner of the review but the name of the parking being reviewed. So I call the same previewcard but with this argument as null to indicate to the function PreviewCard that it's a call from ViewProfileScreen and that i dont need to see the owner's name with his tag and stuff...

File: app/src/main/java/com/github/se/cyrcle/ui/gambling/GamblingScreen.kt (hunk 11 above)
[the remark this answers is not in the document]
Comment (AlexandreMajchrzak) [reply app/src/main/java/com/github/se/cyrcle/ui/gambling/GamblingScreen.kt]: i changed them all one by one in GamblingScreen and in GamblingScreenTest. Took me quite a while because chatgpt is too dumb to do that repetitive task... And i thought AI was gonna steal my job...

File: app/src/main/java/com/github/se/cyrcle/ui/profile/ViewProfileScreen.kt (hunk 10 above)
Comment (LouisJamesVasseur) [reply app/src/main/java/com/github/se/cyrcle/ui/profile/ViewProfileScreen.kt:453]: Ok, fair enough!

File: app/src/main/java/com/github/se/cyrcle/ui/parkingDetails/ParkingDetailsScreen.kt (hunk 6 above)
Comment (AlexandreMajchrzak) [reply app/src/main/java/com/github/se/cyrcle/ui/parkingDetails/ParkingDetailsScreen.kt:494]: Ok it's a bit too complicated for no reason to do that. Because the texts are from different instances (whether material3 or atoms or idk what else  even...), so their arguments differ. The parameters are different in each file.
And instead I ask you to accept my counter-offer, which is that i place the if else condition inside the color argument itself, reducing code size by 2 everywhere. That's already good enough.

Comment (LouisJamesVasseur) [reply app/src/main/java/com/github/se/cyrcle/ui/parkingDetails/ParkingDetailsScreen.kt:494]: Unfortunate that you couldn't get our suggestion to work, but at the same time I understand that it's difficult and it's not really this PR's objective anyway. I'm okay w/ your suggestion, how about you @Armetiis?

**Comment by AlexandreMajchrzak** [comment]
> Hi. I think you did a pretty good job overrall. However, for some reason i still can't see the level of other users (e.g. in the image below, gokunaruto... should be level 15 but I see level 0).
> 
> Also please mind my comment on the duplication of if (color == "rainbow") ... else ... These are pretty big pieces of code that are duplicated (three times). I'd be glad to explain further the implementation i have in mind if my comment was not clear enough.
> 
> image

honestly i cant find anything suspicious that could explain that in my AllReviewsScreen code. But I have an issue with a test in MainActivityTest and it's the getUserById function that is not working. So maybe that's the root of the problem since i call it to get the users' reputations.

**Comment by LouisJamesVasseur** [comment]
> > Hi. I think you did a pretty good job overrall. However, for some reason i still can't see the level of other users (e.g. in the image below, gokunaruto... should be level 15 but I see level 0).
> > Also please mind my comment on the duplication of if (color == "rainbow") ... else ... These are pretty big pieces of code that are duplicated (three times). I'd be glad to explain further the implementation i have in mind if my comment was not clear enough.
> > image
> 
> honestly i cant find anything suspicious that could explain that in my AllReviewsScreen code. But I have an issue with a test in MainActivityTest and it's the getUserById function that is not working. So maybe that's the root of the problem since i call it to get the users' reputations.

Might be able to help. Could you elaborate on what you mean by "not working"?

**Comment by AlexandreMajchrzak** [comment]
> > > Hi. I think you did a pretty good job overrall. However, for some reason i still can't see the level of other users (e.g. in the image below, gokunaruto... should be level 15 but I see level 0).
> > > Also please mind my comment on the duplication of if (color == "rainbow") ... else ... These are pretty big pieces of code that are duplicated (three times). I'd be glad to explain further the implementation i have in mind if my comment was not clear enough.
> > > image
> > 
> > 
> > honestly i cant find anything suspicious that could explain that in my AllReviewsScreen code. But I have an issue with a test in MainActivityTest and it's the getUserById function that is not working. So maybe that's the root of the problem since i call it to get the users' reputations.
> 
> Might be able to help. Could you elaborate on what you mean by "not working"?

look at the CI, there's this test that fails to onSuccess when calling getUserById

File: app/src/main/java/com/github/se/cyrcle/ui/gambling/GamblingScreen.kt
[the remark this answers is not in the document]
Comment (andrepeiry) [reply app/src/main/java/com/github/se/cyrcle/ui/gambling/GamblingScreen.kt:1]: The reason that your CI does not pass is for this : 
```
override fun getUserById(
      userId: String, onSuccess: (User) -> Unit, onFailure: (Exception) -> Unit
  ) {
    db.collection(collectionPath).document(userId).get()
        .addOnSuccessListener { document ->
          val userPublic: UserPublic =
              try {
                Log.d("UserRepositoryFirestore", " Correctly fetched UserPublic")
                deserializeUserPublic(document.data!!)
              } catch (e: Exception) {
                Log.d("UserRepositoryFirestore", "Error fetching UserPublic")
                onFailure(e)                                                          // onFailure
                return@addOnSuccessListener
              }
          db.collection(collectionPath)
              .document(userId).collection("private").document("details").get()
              .addOnSuccessListener { detailsDocument ->
                val userDetails = deserializeUserDetails(detailsDocument.data!!)
                onSuccess(User(userPublic, userDetails))                              // onSuccess
              }
              .addOnFailureListener { onSuccess(User(userPublic, null)) }             // onSuccess
        }
        .addOnFailureListener(onFailure)                                              // onFailure
  }
```
Somehow, your modification in `deserializeUserPublic` does not have the same behavior as before. Maybe try another way to deal with `userReputationScore` that could avoid an error down the line. It is calling `onFailure` instead of `onSuccess`, even though you seems to obtain the good return value.

File: app/src/main/java/com/github/se/cyrcle/model/user/UserRepositoryFirestore.kt [hunk 12]
```diff
@@ -151,7 +151,11 @@ constructor(private val db: FirebaseFirestore, private val auth: FirebaseAuth) :
 
   private fun deserializeUserPublic(data: Map<String, Any>): UserPublic {
     val gson = Gson()
-    return gson.fromJson(gson.toJson(data), UserPublic::class.java)
+    val json = gson.toJson(data)
+    val userPublic = gson.fromJson(json, UserPublic::class.java)
+    // Ensure default values for missing fields
+    val userReputationScore = (data["userReputationScore"] as? Number)?.toDouble() ?: 0.0
+    return userPublic.copy(userReputationScore = userReputationScore)
```
Comment (andrepeiry) [inline app/src/main/java/com/github/se/cyrcle/model/user/UserRepositoryFirestore.kt:158]: The reason that your CI does not pass is for this : 
```
override fun getUserById(
      userId: String, onSuccess: (User) -> Unit, onFailure: (Exception) -> Unit
  ) {
    db.collection(collectionPath).document(userId).get()
        .addOnSuccessListener { document ->
          val userPublic: UserPublic =
              try {
                Log.d("UserRepositoryFirestore", " Correctly fetched UserPublic")
                deserializeUserPublic(document.data!!)
              } catch (e: Exception) {
                Log.d("UserRepositoryFirestore", "Error fetching UserPublic")
                onFailure(e)                                                          // onFailure
                return@addOnSuccessListener
              }
          db.collection(collectionPath)
              .document(userId).collection("private").document("details").get()
              .addOnSuccessListener { detailsDocument ->
                val userDetails = deserializeUserDetails(detailsDocument.data!!)
                onSuccess(User(userPublic, userDetails))                              // onSuccess
              }
              .addOnFailureListener { onSuccess(User(userPublic, null)) }             // onSuccess
        }
        .addOnFailureListener(onFailure)                                              // onFailure
  }
```
Somehow, your modification in `deserializeUserPublic` does not have the same behavior as before. Maybe try another way to deal with `userReputationScore` that could avoid an error down the line. It is calling `onFailure` instead of `onSuccess`, even though you seems to obtain the good return value.

Comment (LouisJamesVasseur) [reply app/src/main/java/com/github/se/cyrcle/model/user/UserRepositoryFirestore.kt:158]: Yes, I can confirm this from looking at the call stack of the failing test. I've been looking into why, still at it...

Comment (LouisJamesVasseur) [reply app/src/main/java/com/github/se/cyrcle/model/user/UserRepositoryFirestore.kt:158]: Well it seems from your new push that the old serializer works for this. I have no objections to reverting as we'll remove the default values for null fields from the deserializers when we clean the database next week, so good work finding a fix!

**Comment by AlexandreMajchrzak** [comment]
please god, make it stop

File: app/src/main/java/com/github/se/cyrcle/ui/gambling/GamblingScreen.kt [hunk 13]
```diff
@@ -35,6 +44,60 @@ import com.github.se.cyrcle.ui.theme.molecules.TopAppBar
...
+ * 5. Text Positioning:
+ *     - Radial text placement using polar coordinates
+ *     - Text radius: 65% of wheel radius (textProportion = 0.65f)
+ *     - Text rotation: Compensates for segment angle + 90° for readability
+ *     - Dynamic scaling: Font size = 6% of wheel radius
+ *
+ * @param modifier Modifier for the composable
+ * @param onSegmentLanded Callback triggered when wheel stops on a segment
+ * @return A function that triggers the wheel spin
+ */
```
Comment (Armetiis) [inline app/src/main/java/com/github/se/cyrcle/ui/gambling/GamblingScreen.kt:100]: Thanks for documenting this!!

File: app/src/androidTest/java/com/github/se/cyrcle/ui/gambling/GamblingScreenTest.kt (hunk 3 above)
Comment (LouisJamesVasseur) [reply app/src/androidTest/java/com/github/se/cyrcle/ui/gambling/GamblingScreenTest.kt:207]: Ok got it, I'll let it go as-is then.

File: app/src/main/java/com/github/se/cyrcle/ui/parkingDetails/ParkingDetailsScreen.kt (hunk 6 above)
Comment (Armetiis) [reply app/src/main/java/com/github/se/cyrcle/ui/parkingDetails/ParkingDetailsScreen.kt:496]: I think we can keep this for now. I suggested that because I was unsure of how complicated it would actually be to make the text rainbow colored. I doubt it's just a "Color" argument. But this is not a concern for now. This can be addressed when actually implementing the rainbow color, as that's when it will be possible to assess if it's necessary or not.

For the moment this is perfect, as the main concern to me was the heavy code duplication.

File: app/src/main/java/com/github/se/cyrcle/ui/parkingDetails/ParkingDetailsScreen.kt (hunk 8 above)
Comment (LouisJamesVasseur) [reply app/src/main/java/com/github/se/cyrcle/ui/parkingDetails/ParkingDetailsScreen.kt:104]: Nice, you changed it!

**Review by LouisJamesVasseur — APPROVED** [review]
You've addressed all the changes I requested and I've found no more bugs. Your PR is approved from me, good work!

**Review by Armetiis — APPROVED** [review]
Good job on fixing everything! I don't think there's much left to say. Approved

File: app/src/main/java/com/github/se/cyrcle/ui/gambling/GamblingScreen.kt
[the remark this answers is not in the document]
Comment (AlexandreMajchrzak) [reply app/src/main/java/com/github/se/cyrcle/ui/gambling/GamblingScreen.kt:1]: Here you go, MASSIVE documentation (and you know what else is massive ? the low taper fade meme)

File: app/src/main/java/com/github/se/cyrcle/ui/parkingDetails/ParkingDetailsScreen.kt (hunk 8 above)
Comment (AlexandreMajchrzak) [reply app/src/main/java/com/github/se/cyrcle/ui/parkingDetails/ParkingDetailsScreen.kt:104]: I deleted that during a rebase when I shouldnt have.

File: app/src/main/java/com/github/se/cyrcle/ui/gambling/GamblingScreen.kt (hunk 13 above)
Comment (AlexandreMajchrzak) [reply app/src/main/java/com/github/se/cyrcle/ui/gambling/GamblingScreen.kt:100]: You're welcome
